modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ParameterMetadata.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.prepare;
+
+import java.util.List;
+import org.apache.ignite.internal.tostring.S;
+
+/**
+ * Metadata for dynamic parameters.
+ */
+public final class ParameterMetadata {
+
+    private final List<ParameterType> parameterTypes;
+
+    /** Constructor. */
+    public ParameterMetadata(List<ParameterType> parameterTypes) {
+        this.parameterTypes = parameterTypes;
+    }
+
+    /** Return metadata for each parameter. */
+    public List<ParameterType> parameterTypes() {
+        return parameterTypes;
+    }
+
+    @Override
+    public String toString() {
+        return S.toString(ParameterMetadata.class, this, "parametersTypes", parameterTypes);
```

Comment:
why do you need additional parameters here?

Reply:
Because `S::toString(this)` always produces `ParameterMetadata []` which does not include string representations of parameters.

Reply:
you can mark `parameterTypes` field with `@IgniteToStringInclude` annotation